libraries/standard/mqtt/src/mqtt.c

```diff
@@ -890,6 +1218,88 @@ static MQTTStatus_t receiveConnack( MQTTContext_t * const pContext,
 
 /*-----------------------------------------------------------*/
 
+static MQTTStatus_t resendPendingAcks( MQTTContext_t * const pContext )
+{
+    MQTTStatus_t status = MQTTSuccess;
+
+    assert( pContext != NULL );
+
+    /* Check if there is any pending acks that need to be resend.
+     * Library owns resending PUBREL for a QoS2 outgoing publish.
+     * PUBREL packet will be resend if there are any packets at state
+     * #MQTTPubCompPending or #MQTTPubRelSend. */
```

Comment:
Why only outgoing publishes? Do we not need to resend PUBACK/PUBREC/PUBCOMP for incoming publishes?

Reply:
Only PUBLISH and PUBREL packets will be resent as per MQTT V3.1.1. PUBLISH will be sent from application.